llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -4111,5 +4143,8 @@ Instruction *InstCombinerImpl::visitSelectInst(SelectInst &SI) {
     }
   }
 
+  if (auto *Instruction = foldSelectToCmp(SI))
```

Comment:
It would be better to move this fold before expensive folds above (e.g., line 4066).